### PR TITLE
Add rxSearchBox

### LINKED ISSUE #844: Add table filtering directive
Add table filtering directive to the framework that has functionality for clearing the filter within the input box (rather than as a separate button which can be mistaken as a submit button). This will make for a cleaner UI and eliminate the issue we're having with table filters currently where "clear" buttons aren't close to the filter box or are mistaken for submit buttons.

@ElementE and I will add mock-ups here soon.

### PR DESCRIPTION
**Pull request by CITguy**
Closes #844

**Defaults / Empty Search Boxes**
screen shot 2015-04-10 at 3 32 52 pm

**Search Boxes with Values**
screen shot 2015-04-10 at 3 33 12 pm

### CODE REVIEW

File: src/rxSearchBox/rxSearchBox.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+angular.module('encore.ui.rxSearchBox', [])
+.directive('rxSearchBox', function () {
+    return {
+        restrict: 'E',
+        require: 'ngModel',
+        templateUrl: 'templates/rxSearchBox.html',
+        scope: {
+            searchVal: '=ngModel',
+            isDisabled: '@ngDisabled',
+            rxPlaceholder: '=?'
```
Comment (parlarjb) [inline src/rxSearchBox/rxSearchBox.js]: What about using the normal HTML `placeholder` attribute?

Comment (CITguy) [inline src/rxSearchBox/rxSearchBox.js]: *(I had a detailed counter argument typed up but I realized that I was contradicting myself. :grin: )*

I've no problem with using `placeholder` for a custom form element. I'll get that changed.

**Comment by glynnis** [comment]
Can you also add a demo to the styleguide? Either to `/styleguide/forms` or `/styleguide/tables`?

Looks like `/styleguide/layouts/2` needs updating as well to remove the "Clear Filter" button and utilize the work you've done here.

File: src/rxSearchBox/rxSearchBox.js (hunk 1 above)
Comment (CITguy) [inline src/rxSearchBox/rxSearchBox.js]: Yeah, so in attempt to use the native attribute, it becomes troublesome to set a default value with two-way binding. After chatting with @parlarjb, we'll leave it as is, for simplicity's sake.

**Comment by CITguy** [comment]
@glynnis: Yes, I'll get styleguide demos in place.

**Comment by CITguy** [comment]
* Updated styleguide with table filtering example and using `rxSearchBox` in Layout 2.
* rebased and squashed

**Comment by CITguy** [comment]
**Layout 2: Unfiltered**
screen shot 2015-04-14 at 2 07 29 pm

**Layout 2: Filtered**
screen shot 2015-04-14 at 2 07 22 pm

**Table With Filtering Demo**
screen shot 2015-04-14 at 2 07 40 pm

File: demo/styleguide/tables.html [hunk 2]
```diff
@@ -40,60 +33,54 @@ <h3 class="title">Table Footer Pagination via rxPaginate</h3>
...
     </ul>
-    </li>
-    <li>With the addition of <a href="http://rackerlabs.github.io/encore-ui/#/component/rxFloatingHeader">rxFloatingHeader</a> to the framework, we now have a standard way of adding table filters. In short, we add a new <code>&lt;tr&gt;</code> into the <code>&lt;thead&gt;</code> of the table. <code>rxFloatingHeader</code> will take care of the styling, by adding the appropriate classes on its own (namely <a href="https://github.com/rackerlabs/encore-ui/blob/master/src/rxFloatingHeader/rxFloatingHeader.less#L12"><code>th.filter-header</code></a>)</li>
-</ul>
 
-<h3 class="title">Nested content</h3>
+    <h3 class="title">Filters</h3>
+    <div class="component-description clear">
+      <ul>
+          <li><span class="msg-info">ENCORE-SPECIFIC:</span> Tables have also been used for filtering. In the past there was no common common design pattern for this, and different products implemented it on their own:
```
Comment (glynnis) [inline demo/styleguide/tables.html]: "In the past there was no common common design pattern for this" --> "In the past there was no ~~common~~ common design pattern for this"

File: demo/styleguide/tables.html [hunk 3]
```diff
@@ -40,60 +33,54 @@ <h3 class="title">Table Footer Pagination via rxPaginate</h3>
...
+          <ul class="list">
+              <li>Data based filters, seen in Encore Cloud (Servers, Volumes, Images, Database, Load Balancers) - Example of server markup</a></li>
+              <li>Ticket Queues (Filtering by ticket queues and status)</li>
+              <li>Billing (Filtering by billing data and transactional status)</li>
+          </ul>
+          </li>
+          <li>With the addition of <a href="http://rackerlabs.github.io/encore-ui/#/component/rxFloatingHeader">rxFloatingHeader</a> to the framework, we now have a standard way of adding table filters. In short, we add a new <code>&lt;tr&gt;</code> into the <code>&lt;thead&gt;</code> of the table. <code>rxFloatingHeader</code> will take care of the styling, by adding the appropriate classes on its own (namely <a href="https://github.com/rackerlabs/encore-ui/blob/master/src/rxFloatingHeader/rxFloatingHeader.less#L12"><code>th.filter-header</code></a>)</li>
```
Comment (glynnis) [inline demo/styleguide/tables.html]: Ah, this raises a good point -- I assume rxSearchBox is all kosher with rxFloatingHeader?

File: demo/styleguide/tables.html [hunk 4]
```diff
@@ -40,60 +33,54 @@ <h3 class="title">Table Footer Pagination via rxPaginate</h3>
...
-    <li>With the addition of <a href="http://rackerlabs.github.io/encore-ui/#/component/rxFloatingHeader">rxFloatingHeader</a> to the framework, we now have a standard way of adding table filters. In short, we add a new <code>&lt;tr&gt;</code> into the <code>&lt;thead&gt;</code> of the table. <code>rxFloatingHeader</code> will take care of the styling, by adding the appropriate classes on its own (namely <a href="https://github.com/rackerlabs/encore-ui/blob/master/src/rxFloatingHeader/rxFloatingHeader.less#L12"><code>th.filter-header</code></a>)</li>
-</ul>
 
-<h3 class="title">Nested content</h3>
+    <h3 class="title">Filters</h3>
+    <div class="component-description clear">
+      <ul>
+          <li><span class="msg-info">ENCORE-SPECIFIC:</span> Tables have also been used for filtering. In the past there was no common common design pattern for this, and different products implemented it on their own:
+          <ul class="list">
+              <li>Data based filters, seen in Encore Cloud (Servers, Volumes, Images, Database, Load Balancers) - Example of server markup</a></li>
```
Comment (glynnis) [inline demo/styleguide/tables.html]: Is there supposed to be a link to the example here?

File: demo/styleguide/tables.html [hunk 5]
```diff
@@ -107,12 +94,10 @@ <h4 class="title sm">Table</h4>
...
 
-<h2 class="title lg" id="roadmap">UI Roadmap / Possible Future-work</h2>
-
-<ul class="list">
-    <li>Checkboxes for table rows with design patterns related to "Select/Unselect All" behavior</li>
-    <li>Loading Styles for Tables</li>
-    <li>Submit actions for Server-Side Filtering</li>
-</ul>
-
+    <h2 class="title lg" id="roadmap">UI Roadmap / Possible Future-work</h2>
```
Comment (glynnis) [inline demo/styleguide/tables.html]: :+1:

File: demo/styleguide/tables.html (hunk 3 above)
Comment (parlarjb) [inline demo/styleguide/tables.html]: And can you update the rxFloatingHeader demo to use rxSearchBox instead of the bare `<input>` it uses now?

Comment (CITguy) [inline demo/styleguide/tables.html]: yup, it's using rx-floating-header in Layout 2

File: demo/styleguide/tables.html (hunk 2 above)
Comment (CITguy) [inline demo/styleguide/tables.html]: (pre-existing text) I'll correct this.

File: demo/styleguide/tables.html (hunk 4 above)
Comment (CITguy) [inline demo/styleguide/tables.html]: (pre-existing text) I'll correct this.

File: demo/styleguide/tables.html (hunk 5 above)
Comment (CITguy) [inline demo/styleguide/tables.html]: (a bit of my O.C.D kicked in with the markup nesting)

File: demo/styleguide/tables.html (hunk 3 above)
Comment (parlarjb) [inline demo/styleguide/tables.html]: Sorry, I specifically meant this: https://github.com/rackerlabs/encore-ui/blob/master/src/rxFloatingHeader/docs/rxFloatingHeader.html#L39

Comment (CITguy) [inline demo/styleguide/tables.html]: Ah, I'll update that.

**Comment by CITguy** [comment]
Updated `rxFloatingHeader` to use `rxSearchBox` per @parlarjb

**Comment by CITguy** [comment]
updated demo markup...

**Comment by parlarjb** [comment]
LGTM

**Comment by CITguy** [comment]
rebased

**Pull request merged**